beacon_node/store/src/hot_cold_store.rs

```diff
@@ -2811,77 +2840,55 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
             "data_availability_boundary" => data_availability_boundary,
         );
 
-        let mut ops = vec![];
-        let mut last_pruned_block_root = None;
+        let mut guard = self.block_cache.lock();
```

Comment:
This lock is being held for a potentially long time while blob pruning runs. I wonder if this could be contributing to some of the slow performance we've seen running `redb`.

I know we're trying to avoid collecting too much junk in memory while pruning, but we should be able to collect all the block roots to delete and then delete these from the cache at the end. Even if we collected 10 years of block roots we'd only use 0.84GB of extra RAM:

```
(225*32*365*10*32)/1e9
```

Reply:
Changed this here to hold block roots in memory https://github.com/sigp/lighthouse/pull/4718/commits/ed9d3472d39d851c4daa018a11e1321c90716ae8